spec/models/spree/order/checkout_spec.rb

```diff
@@ -26,4 +24,32 @@
       end
     end
   end
+
+  describe "order with products with different shipping categories" do
+    let(:order) { create(:order_with_totals_and_distribution, ship_address: create(:address) ) }
+    let(:shipping_method) { create(:shipping_method, distributors: [order.distributor]) }
+    let(:other_shipping_category) { create(:shipping_category) }
+    let(:other_product) { create(:product, shipping_category: other_shipping_category ) }
+    let(:other_variant) { other_product.variants.first }
+
+    before do
+      order.order_cycle = create(:simple_order_cycle,
+                                 distributors: [order.distributor],
+                                 variants: [order.line_items.first.variant, other_variant])
+      order.line_items = [order.line_items.first, create(:line_item,
```

Comment:
What about `order.line_items << create(:line_item, order: order, variant: other_variant)`?. AR is here to help you 😂

Reply:
gosh, why did I do that? 🤣 
I fixed it in a new commit.